### DISCUSSION TITLE
Neuropixels / Imec module

### DISCUSSION THREAD

**Discussion by vncntprvst** [bonsai-rx/bonsai#1254] [Q&A] [answered]
This is a recurrent question topic on the Google forum and Discord, but is there any plan to update the Neuropixels module? 
Correct me if I'm wrong, the ONIX module is not compatible with PXI chassis configuration. On the other hand, the old [Neuropixels module](https://github.com/kampff-lab/bonsai.neuropix) does not seem to work with NPX 1.0 and beyond (it also only works with Bonsai-32bits apparently). 
There is a number of ways where having a Bonsai module for Neuropixels probes would simplify development and better integrate with experimental workflows.

**Comment by glopesdev** [comment, marked as answer]
I very much agree. To be honest this is mostly a matter of time and current priorities, we haven't seen a lot of interest from the community in this so far and we have been focusing on getting other packages up and running.

Another issue is clarifying what is the licensing situation re. distribution and use of the neuropix API. I see that SpikeGLX distributes header files and a .lib binary from IMEC but there is no specific license file there other than the copyright statement from IMEC.

Ideally we like to make sure we can distribute self-contained packages to create a plug-and-play experience so before we invest too much time we would need to sit down and resolve the practicalities of this.

I suspect it will happen in time, but currently cannot give a reliable forecast unfortunately.

**Reply by vncntprvst** [reply]
That makes sense. Thanks for clarifying the situation.

**Reply by jbscherrer** [reply]
Are there any recent developments on this front? Alternatively, are there standard ways to pipe Neuropixels data from e.g. Open Ephys interface into Bonsai?

**Reply by glopesdev** [reply]
@jbscherrer Given the sheer volume of data I suspect any piping of raw NPX data across processes might cause a performance bottleneck. If it's maybe just a couple of channels, then you might be able to use the [Ephys Socket plugin](https://open-ephys.github.io/gui-docs/User-Manual/Plugins/Ephys-Socket.html).

Finally, another option if you have neuropixel headstages would be to use the new ONIX system which has a built-in [Bonsai interface](https://open-ephys.github.io/onix-docs/Software%20Guide/Bonsai.ONIX/index.html).
